Kryptonite/Request.swift

```diff
@@ -115,18 +113,8 @@ struct SignRequest:Jsonable {
 
         (session, user, digestType) = try SignRequest.parse(requestData: data)
         
-        do {
-            let potentialHostAuth = try HostAuth(json: json ~> "host_auth")
-            
-            guard try potentialHostAuth.verify(sessionID: session) == true
-            else {
-                throw InvalidHostAuthSignature()
-            }
-
-            self.hostAuth = potentialHostAuth
-        } catch {
-            log("host auth error: \(error)")
-            hostAuth = nil
+        if let potentialHostAuth = try? HostAuth(json: json ~> "host_auth") {
+            verifiedHostAuth = try? VerifiedHostAuth(session: session, hostAuth: potentialHostAuth)
```

Comment:
is there an easy way to have the JSON initializer call init so the same hostAuth logic is used for each?